src/java.base/share/classes/java/lang/FdLibm.java

```diff
@@ -504,6 +504,451 @@ static double compute(double y, double x) {
+                }
+                m -= i-1;
+                ix0 |= (ix1 >>> (32 - i)); // unsigned shift
+                ix1 <<= i;
+            }
+            m -= 1023;      // unbias exponent */
+            ix0 = (ix0 & 0x000f_ffff) | 0x0010_0000;
+            if ((m & 1) != 0){        // odd m, double x to make it even
+                ix0 += ix0 + ((ix1 & sign) >>> 31); // unsigned shift
+                ix1 += ix1;
+            }
+            m >>= 1;        // m = [m/2]
+
+            // generate sqrt(x) bit by bit
+            ix0 += ix0 + ((ix1 & sign) >>> 31); // unsigned shift
+            ix1 += ix1;
+            q = q1 = s0 = s1 = 0;   // [q,q1] = sqrt(x)
+            r = 0x0020_0000;        // r = moving bit from right to left
+
+            while(r != 0) {
+                t = s0 + r;
+                if(t <= ix0) {
+                    s0   = t + r;
+                    ix0 -= t;
+                    q   += r;
+                }
+                ix0 += ix0 + ((ix1 & sign) >>> 31); // unsigned shift
+                ix1 += ix1;
+                r >>>= 1; // unsigned shift
+            }
+
+            r = sign;
+            while (r != 0) {
+                t1 = s1 + r;
+                t  = s0;
+                if((t < ix0) ||
+                   ((t == ix0) && (Integer.compareUnsigned(t1, ix1) <= 0 ))) { // t1 <= ix1
+                    s1  = t1 + r;
+                    if (((t1 & sign) == sign) && (s1 & sign) == 0) {
+                        s0 += 1;
+                    }
+                    ix0 -= t;
+                    if (Integer.compareUnsigned(ix1, t1) < 0) {  // ix1 < t1
+                        ix0 -= 1;
+                    }
+                    ix1 -= t1;
+                    q1  += r;
+                }
+                ix0 += ix0 + ((ix1 & sign) >>> 31);
+                ix1 += ix1;
+                r >>>= 1; // unsigned shift
+            }
+
+            // use floating add to find out rounding direction
+            if ((ix0 | ix1) != 0) {
+                z = 1.0 - tiny; // trigger inexact flag
+                if (z >= 1.0) {
+                    z = 1.0 + tiny;
+                    if (q1 == 0xffff_ffff) {
+                        q1 = 0;
+                        q += 1;
+                    } else if (z > 1.0) {
+                        if (q1 == 0xffff_fffe) {
+                            q += 1;
+                        }
+                        q1 += 2;
+                    } else {
+                        q1 += (q1 & 1);
+                    }
+                }
+            }
+            ix0 = (q >> 1) + 0x3fe0_0000;
+            ix1 =  q1 >>> 1; // unsigned shift
+            if ((q & 1) == 1) {
+                ix1 |= sign;
+            }
+            ix0 += (m << 20);
+            z = __HI(z, ix0);
+            z = __LO(z, ix1);
+            return z;
+        }
+    }
+
+    // The following comment is supplementary information from the FDLIBM sources.
+
+    /*
```

Comment:
I didn't think it was necessary to preserve this supplemental information in both the transliteration port and the mainline sources.